internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -538,13 +537,16 @@ func (c *FunctionComposer) Compose(ctx context.Context, xr *composite.Unstructur
 	k := xr.GetKind()
 	n := xr.GetName()
 	u := xr.GetUID()
+	r := xr.GetCompositionReference()
```

Comment:
this file is the only difference from https://github.com/crossplane/crossplane/pull/6683 I believe - do we need to make these changes also in main and release-2.0?

Reply:
Hey @jbw976, you're right. I was biased about the function pipeline but it turned out that, the values were overwritten by the claim syncer. I updated the last commit and opened a [PR](https://github.com/crossplane/crossplane/pull/6881) to main branch as well.